### ISSUE TITLE
lint: fix sqlRowsClose checker panic

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#537]
If we add this function to sanity checks:
```go

func sliceExpressions(xs []int) {
	_ = xs[:]
	_ = xs[0:]
	_ = xs[:0]
	_ = xs[0:0]
	_ = xs[:0:0]
	_ = xs[0:0:0]
}
```
`sqlRowsClose` checker starts to fail them:
```
        all_checkers_test.go:58: unexpected panic: `runtime error: invalid memory address or nil pointer dereference`
            goroutine 209 [running]:
            runtime/debug.Stack(0xc0001efba0, 0x7f7e20, 0xb9f500)
            	$GOROOT/src/runtime/debug/stack.go:24 +0xa7
            github.com/go-critic/go-critic/lint.TestSanity.func1.1(0xc000474a00, 0xc000154820, 0xc0000ae170)
            	$GOPATH/src/github.com/go-critic/go-critic/lint/all_checkers_test.go:58 +0x123
            panic(0x7f7e20, 0xb9f500)
            	$GOROOT/src/runtime/panic.go:494 +0x1b9
            github.com/go-critic/go-critic/lint.(*sqlRowsCloseChecker).VisitFuncDecl(0xc0000afed0, 0xc0003fb050)
            	$GOPATH/src/github.com/go-critic/go-critic/lint/sqlRowsClose_checker.go:49 +0x6cd
            github.com/go-critic/go-critic/lint/internal/astwalk.(*funcDeclWalker).WalkFile(0xc00016fe40, 0xc0003dd080)
            	$GOPATH/src/github.com/go-critic/go-critic/lint/internal/astwalk/func_decl_walker.go:15 +0xd7
            github.com/go-critic/go-critic/lint.(*Checker).Check(0xc000322be0, 0xc0003dd080, 0x0, 0xc000322be0, 0xc0000ae170)
            	$GOPATH/src/github.com/go-critic/go-critic/lint/lint.go:95 +0x46
            github.com/go-critic/go-critic/lint.TestSanity.func1(0xc000474a00)
            	$GOPATH/src/github.com/go-critic/go-critic/lint/all_checkers_test.go:64 +0x28d
            testing.tRunner(0xc000474a00, 0xc00022b7a0)
            	$GOROOT/src/testing/testing.go:827 +0xbf
            created by testing.(*T).Run
            	$GOROOT/src/testing/testing.go:878 +0x353
```
